sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -218,6 +237,92 @@ void gnrc_rpl_recv_DIS(gnrc_rpl_dis_t *dis, ipv6_addr_t *src, ipv6_addr_t *dst,
     }
 }
 
+static bool _gnrc_rpl_check_options_validity(int msg_type, gnrc_rpl_dodag_t *dodag,
```

Comment:
Wouldn't it be more efficient to save the while loop here and call the check instead inside the loop in `_pars_options()`?